guide/how-it-works/ecash/introduction.md

```diff
@@ -143,7 +141,21 @@ The process of creating bitcoin backed ecash is known as **minting.** The proces
    height = 456
    layout = "full-width"
 %}
-Sending ecash and receiving can be done via multiple methods to suit different transaction needs. One of the advantages of ecash is that users are simply transacting pieces of data. These pieces of data are very flexible and can be transmitted via any medium or rail. Some unique ways that ecash can be sent and received, which are not possible with on-chain or Lightning transactions, include:
+When a user sends an ecash token to another user, the wallet verifies the mint’s signature to confirm the token’s authenticity. The mint then checks its database to ensure the token has not been previously spent. If the token is valid, it is destroyed and replaced with a new token, which is then issued to the recipient. This cycle of destroying the old token and issuing a new one prevents double spending while maintaining user privacy and security.
```

Comment:
Nice, like this replacement! How does this work when internet connectivity is absent?

Reply:
Ty! 

Without internet connectivity users can still send and receive Cashu tokens via airdrop or text message, scanning QR codes, etc.. but that's just sending and receiving. Technically the claimant needs to be online to **redeem** the ecash token with the mint. However, some wallet display a warning that they are currently offline (no internet connectivity) and informs the user that the token will be claimed once the wallet is back online. The wallet then holds the token in a queue, where it usually appears in the transaction history as ‘pending’ or ‘received offline.’ Once the wallet is back online, it attempts to redeem the ecash.

There are risks with this approach. The most obvious is that while the token hasn’t been redeemed, it could technically be double-spent. If the sender sends that token to someone else or claims it themselves before the claimant, it will become invalid.

Reply:
In cashu a wallet can only verify a mint signature if the proof also includes a [DLEQ proof](https://github.com/cashubtc/nuts/blob/main/12.md). If the token does not include proofs with a valid DLEQ proof and a P2PK locked secret to the receiver there is a risk the sender can double spent before the receiver can swap with the mint, if those two conditions are met there is not a risk of a double spent.

Reply:
I think I follow: In the case of offline sending the best way to reduce the risk of double spending is to ensure that a token includes both a valid DLEQ proof and is P2PK locked to the recipient. In terms of UX, the wallet automatically handles the inclusion of DLEQ proofs when sending ecash tokens, even without a P2PK lock. However, a token with just a DLEQ proof and no P2PK lock is still at risk of being double spent.

Some ways it could be double spent (according to Claude):

1. Proof of Possession: DLEQ proofs demonstrate that the sender possessed a valid token at the time of creating the proof, but they don't prevent the sender from using the same token elsewhere before the recipient redeems it.
2. Race Condition: Without additional measures (like P2PK locks), there's a potential race condition where the sender could attempt to spend the token with the mint before the recipient does.

I think this might be too technical for inclusion in this guide. The target audience is designers who are familiar with Bitcoin but want to learn about ecash from a design perspective. In this case, I think we might be able to avoid getting into this level of detail. What do you think?